app/Commands/GenerateCommand.php

```diff
@@ -30,13 +32,138 @@ class GenerateCommand extends Command
-        ]));
+        ];
+
+        // define the list of templates to render
+        $templates = [
+            'dockerfile' => 'Dockerfile',
+        ];
+
+        // ... add additional templates here, possibly based on scanning the source ...
+
+        // render each template
+        foreach ($templates as $template => $output) {
+            $r = $this->writeTemplateFile($template, $options, $output);
+
+            if ($r === false) {
+                return Command::FAILURE;
+            }
+        }
+
+        return Command::SUCCESS;
+    }
+
+    /**
+     * Write a template file to the filesystem
+     *
+     * @param string $template
+     * @param array $options
+     * @param string $output
+     * @return bool
+     */
+    public function writeTemplateFile($template, $options, $output) {
+        // read the file before the change
+        try {
+            if (file_exists($output)) {
+              $before = file($output, FILE_IGNORE_NEW_LINES);
+
+                if ($before === false) {
+                    $before = [];
+                }
+            } else {
+                $before = [];
+            }
+        } catch (Exception $e) {
+            $before = [];
+        }
+
+        // render the template
+        $result = explode("\n", (string)view($template, $options));
+        if (end($result) === '' && end($before) !== '') {
+            array_pop($result);
+        }
+
+        // write the file if it doesn't exist; if it has changed ask the user what to do
+        if (empty($before)) {
+            echo $this->green(str_pad('create', 11, ' ', STR_PAD_BOTH)) . ' ' . $output . PHP_EOL;
+            file_put_contents($output, implode("\n", $result) . "\n");
+        } elseif ($before === $result) {
+            echo $this->blue(str_pad('identical', 11, ' ', STR_PAD_BOTH)) . ' ' . $output . PHP_EOL;
+        } elseif ($this->answer === 'N') {
+            echo $this->blue(str_pad('skipped', 11, ' ', STR_PAD_BOTH)) . ' ' . $output . PHP_EOL;
+        } else {
+            if ($this->answer !== 'a') {
+                echo $this->red(str_pad('conflict', 11, ' ', STR_PAD_BOTH)) . ' ' . $output . PHP_EOL;
+            }
+
+            while (true) {
+                if ($this->answer !== 'a') {
+                    $this->answer = readline('Overwrite ' . $output . '? (enter "h" for help) [Ynaqdh] ');
+                    $this->answer = strtolower($this->answer);
+                }
+
+                if ($this->answer === 'y' || $this->answer === '' || $this->answer === 'a') {
+                    echo $this->yellow(str_pad('forced', 11, ' ', STR_PAD_BOTH)) . ' ' . $output . PHP_EOL;
+                    return file_put_contents($output, implode("\n", $result) . "\n");
+                } elseif ($this->answer === 'n') {
+                    return true;
+                } elseif ($this->answer === 'd') {
+                    $diff = new \Diff($before, $result);
```

Comment:
Answering with `d` throws an error `Class "Diff" not found` like so:
```
  Class "Diff" not found

  at app/Commands/GenerateCommand.php:124
    120▕                     return file_put_contents($output, implode("\n", $result) . "\n");
    121▕                 } elseif ($this->answer === 'n') {
    122▕                     return true;
    123▕                 } elseif ($this->answer === 'd') {
  ➜ 124▕                     $diff = new \Diff($before, $result);
    125▕                     $renderer = new \Diff_Renderer_Text_Unified();
    126▕                     foreach (explode("\n", $diff->render($renderer)) as $line) {
    127▕                         if (str_starts_with($line, '---') || str_starts_with($line, '+++')) {
    128▕                             echo line . PHP_EOL;

  1   app/Commands/GenerateCommand.php:59
      App\Commands\GenerateCommand::writeTemplateFile()
      +13 vendor frames 

  15  dockerfile-laravel:35
      LaravelZero\Framework\Kernel::handle()
  ```

Reply:
Oh nevermind! I needed to do a composer install to install the dependency for that class

Reply:
I think because `composer.json` changed, you need to run `composer install` (or perhaps it is `composer update`, I'm not all that familiar with composer).  See:

https://github.com/fly-apps/dockerfile-laravel/pull/19/commits/3703776fc9e42ffa27f611a481ca4097d10b51cf#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34